src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.ts

```diff
@@ -0,0 +1,229 @@
+/** Provides the static, procedural versions of the functions provided in the UBAFeeSpokeCalculator */
+
+import { BigNumber } from "ethers";
+import { UbaFlow, isUbaInflow } from "../interfaces";
+import { TokenRunningBalanceWithNetSend, UBAActionType, UBAFlowFee } from "./UBAFeeTypes";
+import UBAConfig from "./UBAFeeConfig";
+import { toBN } from "../utils";
+import { computePiecewiseLinearFunction } from "./UBAFeeUtility";
+import { analog } from ".";
+
+/**
+ * Calculates the running balance for a given token on a spoke chain
+ * @param flows The flows to calculate the running balance for
+ * @param lastValidatedRunningBalance The last validated running balance for the token
+ * @param lastValidatedIncentiveRunningBalance The last validated incentive running balance for the token
+ * @param chainId The chain id of the spoke chain
+ * @param tokenSymbol The token symbol of the token to calculate the running balance for
+ * @param config The UBAConfig to use for the calculation
+ * @returns The running balance for the token
+ */
+export function calculateHistoricalRunningBalance(
+  flows: UbaFlow[],
+  lastValidatedRunningBalance: BigNumber,
+  lastValidatedIncentiveRunningBalance: BigNumber,
+  chainId: number,
+  tokenSymbol: string,
+  config: UBAConfig
+): TokenRunningBalanceWithNetSend {
+  // Attempt to resolve the trigger hurdle to include in the running
+  // balance calculation
+  const { upperBound: upperBoundTriggerHurdle, lowerBound: lowerBoundTriggerHurdle } =
+    config.getBalanceTriggerThreshold(chainId, tokenSymbol);
```

Comment:
Wouldn't you need to pass in `blockNumber` into this call to figure out which trigger threshold to pull from the config store? The threshold can be updated between flows so you want the correct one

Reply:
That's fair - my understanding with this function is that it this function computes the running balances of a current bundle. For efficiency, the assumption is made that the config does not change during intra-bundle calculations - I believe we discussed this and my understanding is that this is an acceptable assumption.

Because of that, the `UBAConfig` is fixed for the calculations of each flow.

Reply:
Resolved offline